### ISSUE TITLE
App closes unexpectedly when navigating to Home

### ISSUE THREAD

**Issue by ShridharGoel** [commons-app/apps-android-commons#2200]
**Summary:** 

App closes unexpectedly when navigating to Home from Explore.

**Steps to reproduce:** 

1. Go to Explore from the navigation drawer.
2. Again open the navigation drawer and click on Explore.
3. Now, click on Home in the navigation drawer and the app will close.

**Device and Android version:** 

Android 7.1.2 (API 25)

**Commons app version:** 

2.9.0
Branch: master
Build variant: betaDebug

**GIF:** 

![ezgif com-video-to-gif 11](https://user-images.githubusercontent.com/35566748/50303681-da526200-04b3-11e9-896c-ab7d1951e629.gif)

**Would you like to work on the issue?**

Yes

**Comment by domdomegg** [comment]
Cannot reproduce on `betaDebug` on `master` @ f521e8886dcf9a2be6ab5848cfb050bbb8acb65a

![api28](https://user-images.githubusercontent.com/4953590/50304371-aeb98280-0487-11e9-8ce7-d6a511908769.gif)

Do you have any logs? Exactly what version of the app are you using? Did you build it with Android Studio, or is it from the Google Play alpha/beta/prod channel?

**Comment by domdomegg** [comment]
The above was recorded on a `Galaxy Nexus (emulator)` at API level `28`.

Also can't reproduce the crash at API level `25`.

**Comment by ShridharGoel** [comment]
App isn't crashing, it's just closing.

**Comment by domdomegg** [comment]
Managed to recreate at 340b0812 at API level `25`.

Seems to happen with any activity in the navbar (tested bookmarks, notifications)

**Comment by domdomegg** [comment]
Another clue: App exits if you press 'back' button instead of navigating home.

**Comment by ShridharGoel** [comment]
Yes, happens with other activities too as I've mentioned in the summary. 

I have almost fixed it.

**Comment by domdomegg** [comment]
Actually, this looks like a bug in Android itself. Maybe a workaround in one of the comments here:

https://issuetracker.google.com/issues/36986021#c2

**Comment by domdomegg** [comment]
It is a bug confirmed specifically in this comment:
https://issuetracker.google.com/issues/36986021#comment7
I think it affects 4.4 (API level 25) only.

I think I've found a workaround for this issue. Calling it with `FLAG_ACTIVITY_SINGLE_TOP` avoids the extra intent call, which in turns fixes this issue without losing the other sensible back behaviour.

However I've there's another bug on API level `25` which is caused by the same bug in Android.
1. Open explore
2. Open bookmarks
3. Open explore
4. Press back

**Comment by ShridharGoel** [comment]
Yes I saw this. After spending considerable time in trying to figure out the reason behind the behaviour, I read about this issue and found out that it's a known Android bug which hasn't been fixed from many years. It's there only in some Android versions like Android 4.4 (API 19), Android 7.0 (API 24), Android 7.1.1 (API 25). Hopefully, the workaround will fix it.
